coreclient/src/conversations/persistence.rs

```diff
@@ -88,41 +106,62 @@ impl Conversation {
         group_id: &GroupId,
     ) -> Result<Option<Conversation>, rusqlite::Error> {
         let group_id = GroupIdRefWrapper::from(group_id);
-        let mut stmt = connection.prepare("SELECT conversation_id, conversation_title, conversation_picture, group_id, last_read, conversation_status, conversation_type FROM conversations WHERE group_id = ?")?;
+        let mut stmt = connection.prepare(
+            "SELECT
+                conversation_id,
+                conversation_title,
+                conversation_picture,
+                group_id,
+                last_read,
+                conversation_status,
+                conversation_type
+            FROM conversations
+            WHERE group_id = ?",
+        )?;
         stmt.query_row(params![group_id], Self::from_row).optional()
     }
 
     pub(crate) fn load_all(connection: &Connection) -> Result<Vec<Conversation>, rusqlite::Error> {
-        let mut stmt = connection.prepare("SELECT conversation_id, conversation_title, conversation_picture, group_id, last_read, conversation_status, conversation_type FROM conversations")?;
+        let mut stmt = connection.prepare(
+            "SELECT
+                conversation_id,
+                conversation_title,
+                conversation_picture,
+                group_id,
+                last_read,
+                conversation_status,
+                conversation_type
+            FROM conversations",
+        )?;
         let rows = stmt.query_map([], Self::from_row)?;
         rows.collect()
     }
 
-    pub(super) fn update_conversation_picture(
-        &self,
+    pub(super) fn update_picture(
         connection: &Connection,
         notifier: &mut StoreNotifier,
+        conversation_id: ConversationId,
         conversation_picture: Option<&[u8]>,
     ) -> rusqlite::Result<()> {
         connection.execute(
             "UPDATE conversations SET conversation_picture = ? WHERE conversation_id = ?",
-            params![conversation_picture, self.id],
+            params![conversation_picture, conversation_id],
```

Comment:
What was wrong with `self.id`?

Reply:
There was a small asymmetry. On one hand we pass the Conversation as &self which contains the picture, on the other hand we pass the data for the new picture as a parameter. From the function signature it is then not entirely obvious which data is actually used. Also, only the id is actually needed nothing else from the conversation.

There is a similar change in UserProfile.